### PR TITLE
#163271060 Build React Components to Complete Signup Process 

### PR DESCRIPTION
**Pull request by Daymorelah**
#### What does this PR do?
Allows a user to finish the signup/registration process
#### Description of Task to be completed?
* Build component to handle confirmation of email address.
* Update Redux store with user-info gotten from the backend
* Write tests for the components, Reducers and actions
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/163271060

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -71,7 +72,9 @@
     "jquery": "3.3.1",
     "node-sass": "4.11.0",
     "react-test-renderer": "16.7.0",
+    "redux-mock-store": "^1.5.3",
```
Comment (kingisaac95) [inline package.json]: Lock version @Daymorelah

File: package.json [hunk 2]
```diff
@@ -71,7 +72,9 @@
...
     "sass-loader": "7.1.0",
+    "sinon": "^7.2.3",
```
Comment (kingisaac95) [inline package.json]: Ditto

File: src/index.html [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+<html lang="en">
   <head>
-    <meta charset="utf-8">
+    <meta charset="utf-8"> 
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
-    <link 
-      rel="stylesheet" 
-      href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" 
-      crossorigin="anonymous">
+    <link rel="stylesheet"
```
Comment (kingisaac95) [inline src/index.html]: One attribute per line. Take `rel` to the next line

File: src/styles/base/_base.scss [hunk 4]
```diff
@@ -2,4 +2,7 @@
 body {
   background-color: $light-purple;
   font-family: $reem-kufi;
+  margin: 0;
+  padding: 0;
+  max-height: 100vh;
```
Comment (kingisaac95) [inline src/styles/base/_base.scss]: Why do we have this style? I don't think we want to set that

File: src/styles/base/_settings.scss [hunk 5]
```diff
@@ -12,3 +12,4 @@ $small-space: 10px;
 $large-ratio: 100%;
 $reem-kufi: 'Reem Kufi', sans-serif;
 $Xlarge-ratio: 300%;
+$font-Family: 'Reem Kufi', sans-serif;
```
Comment (kingisaac95) [inline src/styles/base/_settings.scss]: Newline here

File: src/styles/components/_SignupPage.scss [hunk 6]
```diff
@@ -26,16 +26,16 @@ h2 {
 }
 
 .signup-form-input::placeholder { /* Chrome/Opera/Safari */
-  color: $purple;
+  color: #c4c4c4;
```
Comment (kingisaac95) [inline src/styles/components/_SignupPage.scss]: Why did you remove the variable? Put it back please

File: package.json (hunk 1 above)
Comment (Daymorelah) [reply package.json]: I have locked their versions

File: src/tests/actions/signupActions.test.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    const userDetails = {
+      username: 'someUsername',
+      password: '34565432',
+      email: 'someEmail',
+      firstName: 'someFirstName',
+      lastName: 'someLastName'
+    };
+    it('should complete user signup registration', async () => {
+      const stubPostMethod = sinon.stub(axios, 'post').returns(userInfo);
+      const expectedActions = [];
```
Comment (kingisaac95) [inline src/tests/actions/signupActions.test.js]: You don't need this variable. You can simply say `expect(store.getActions()).toEqual([]);`

File: src/utilities/apiRequests.js [hunk 8]
```diff
@@ -8,7 +8,7 @@ export const Get = async (route) => {
     const response = await axios.get(baseUrl + route);
     return response.data;
   } catch (error) {
-    return error.response.data;
+    return error.response ? error.response.data : error;
```
Comment (kingisaac95) [inline src/utilities/apiRequests.js:11]: Don't we have an error handler? Or are the errors already handled somewhere?

File: src/index.html (hunk 3 above)
Comment (Daymorelah) [reply src/index.html]: I have moved it to a new line.

File: package.json (hunk 2 above)
Comment (Daymorelah) [reply package.json]: I have locked the version

File: src/styles/base/_base.scss (hunk 4 above)
Comment (Daymorelah) [reply src/styles/base/_base.scss]: It has been removed.

File: src/styles/base/_settings.scss (hunk 5 above)
Comment (Daymorelah) [reply src/styles/base/_settings.scss]: A new line has been added.

File: src/styles/components/_SignupPage.scss (hunk 6 above)
Comment (Daymorelah) [reply src/styles/components/_SignupPage.scss]: The Product Owner wanted the placeholder to be hash in colour. I will use a variable instead of the hex value.

Comment (Daymorelah) [reply src/styles/components/_SignupPage.scss]: I have used a variable.

File: src/tests/actions/signupActions.test.js (hunk 7 above)
Comment (Daymorelah) [reply src/tests/actions/signupActions.test.js]: The unnecessary array variable has been removed.

File: src/utilities/apiRequests.js (hunk 8 above)
Comment (Daymorelah) [reply src/utilities/apiRequests.js:11]: We are handling it in the action creators.

**Review by kingisaac95 — APPROVED** [review]
